tapchannel/commitment.go

```diff
@@ -680,6 +724,7 @@ func CreateAllocations(chanState *channeldb.OpenChannel, ourBalance,
 			Type:           allocType,
 			Amount:         rfqmsg.Sum(htlc.AssetBalances),
 			AssetVersion:   asset.V1,
+			SplitRoot:      shouldHouseSplitRoot,
```

Comment:
If the initiator has any asset (either as a channel balance or as an HTLC in their direction) then that allocation gets the split root (in case of multiple HTLCs, just the first). If there is no balance for the initiator (all settled towards the responder), then there is no split root and it's a full value send. No zero value tombstone needed because all outputs are marked as interactive.